var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -747,11 +735,10 @@ def configure_args(self):
             lustre_opt = '--with-lustre={0}'.format(spec['lustre'].prefix)
             config_args.append(lustre_opt)
         # External libevent/pmix
-        if spec.satisfies('@5:') or spec.satisfies('+pmix'):
+        if '^pmix' in spec:
+            # Externally supplied PMIx
             config_args.append('--with-pmix={0}'.format(spec['pmix'].prefix))
             config_args.append('--with-libevent={0}'.format(spec['libevent'].prefix))
```

Comment:
```suggestion
        if '^pmix' in spec:
            config_args.append('--with-pmix={0}'.format(spec['pmix'].prefix))
        if '^libevent' in spec:
            config_args.append('--with-libevent={0}'.format(spec['libevent'].prefix))
```

Reply:
This will still set `--with-libevent` when pmix is not vendored. I hope OpenMPI ignores the flag since it's not its direct dep?